src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -233,6 +228,13 @@ public void RequestHeartbeat()
             _monitor.RequestHeartbeat();
         }
 
+        private bool ContainsTimeOutException(MongoConnectionException connectionException)
```

Comment:
I would rename this to `ContainsSocketTimeOutException`

Reply:
This logic has been moved to a property in  `MongoConnectionException` per Dima's suggestion :)